<a href="https://colab.research.google.com/github/fatimansilla95-svg/PT1_Mineria_Datos/blob/main/Notebooks/01_limpieza_datos.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Proyecto Integrador - Minería de Datos 1**

**Dataset: streaming_users_dirty.json**


Objetivo: conocer la estructura y calidad del dataset.

**Importación de librerías**

In [ ]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
from sklearn.preprocessing import StandardScaler
from sklearn.decomposition import PCA

**Carga del dataset**

In [ ]:
import pandas as pd

df = pd.read_json("/content/sample_data/streaming_users_dirty.json")

In [ ]:
!ls -F /content/sample_data/

anscombe.json*		      mnist_test.csv	     streaming_users_dirty.json
california_housing_test.csv   mnist_train_small.csv
california_housing_train.csv  README.md*


**Inspección inicial**

In [ ]:
print("Información del dataset:")
df.info()


Información del dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8160 entries, 0 to 8159
Data columns (total 8 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   user_id                   8160 non-null   int64  
 1   age                       8160 non-null   int64  
 2   subscription_plan         8160 non-null   object 
 3   monthly_watch_time_mins   7967 non-null   float64
 4   country                   8160 non-null   object 
 5   favorite_genre            7920 non-null   object 
 6   last_login_date           7840 non-null   object 
 7   customer_support_tickets  8160 non-null   int64  
dtypes: float64(1), int64(3), object(4)
memory usage: 510.1+ KB


In [ ]:
df.describe(include='all')


,user_id,age,subscription_plan,monthly_watch_time_mins,country,favorite_genre,last_login_date,customer_support_tickets
count,8160.000000,8160.000000,8160,7967.000000,8160,7920,7840,8160.000000
unique,NaN,NaN,15,NaN,26,28,3062,NaN
top,NaN,NaN,Básico,NaN,Brasil,Comedia,2026-15-03,NaN
freq,NaN,NaN,3450,NaN,1132,1112,20,NaN
mean,13995.433824,34.096814,NaN,1107.346153,NaN,NaN,NaN,1.800980
std,2310.810660,14.511304,NaN,5310.442604,NaN,NaN,NaN,11.334969
min,10000.000000,-5.000000,NaN,-120.000000,NaN,NaN,NaN,-1.000000
25%,11987.750000,25.000000,NaN,489.200000,NaN,NaN,NaN,0.000000
50%,13998.500000,33.000000,NaN,757.400000,NaN,NaN,NaN,1.000000
75%,15997.250000,42.000000,NaN,1045.700000,NaN,NaN,NaN,1.000000


En la inspección inicial del dataset streaming_users_dirty.json se realizó utilizando el método df.info(), que permite obtener una visión general del conjunto de datos. Este comando muestra información esencial como:

Cantidad de registros: 8160 filas, lo que indica el tamaño del dataset.

Número de columnas: 8 variables en total.

Tipos de datos: Se se identificaron variables numéricas(int64, float64)(edad, horas de uso, tiempo de suscripción) y categóricas (object)plataforma, país, género), lo cual es útil para decidir qué tipo de análisis aplicar.

Valores nulos: Se detectaron valores faltantes en las columnas “edad” y “horas_uso”, y algunos registros duplicados. (monthly_watch_time_mins, favorite_genre, last_login_date), lo que sugiere la necesidad de limpieza o imputación antes del análisis exploratorio
